### PR TITLE
Make facet indexation and storage optional

### LINKED ISSUE #986: Facet being stored should be optional
Right now facet field are always stored. It should be optional.
Arguably, indexed should be optional too.

### PR DESCRIPTION
**Pull request by lpouget**
Added a `FacetOptions` for `HierarchicalFacet` which add indexed and stored flags to it.
Propagate change and update tests accordingly
Added a test to ensure that a not indexed flag was taken care of.
Added on `Value` implementation the `path()` function to return the stored facet.

Fixes #986

### CODE REVIEW

File: src/schema/field_entry.rs [hunk 1]
```diff
@@ -132,7 +133,7 @@ impl FieldEntry {
             | FieldType::Date(ref options) => options.is_stored(),
             FieldType::Str(ref options) => options.is_stored(),
             // TODO make stored hierarchical facet optional
```
Comment (fmassot) [inline src/schema/field_entry.rs]: I think you can now remove this comment :)

**Review by fmassot — COMMENTED** [review]
This PR seems good, I still need to understand the tests test_facet_not_populated_for_any_docs / test_facet_not_indexed_for_all_docs, not sure what is the goal here

**Comment by lpouget** [comment]
> This PR seems good, I still need to understand the tests test_facet_not_populated_for_any_docs / test_facet_not_indexed_for_all_docs, not sure what is the goal here

The goal here was to be able to ensure presence of the doc in the store even if it's not indexed (and by extension check it's not indexed).

But maybe it's not the best place to do this.

File: src/core/segment_reader.rs [hunk 2]
```diff
@@ -108,19 +108,22 @@ impl SegmentReader {
     /// Accessor to the `FacetReader` associated to a given `Field`.
     pub fn facet_reader(&self, field: Field) -> crate::Result<FacetReader> {
         let field_entry = self.schema.get_field_entry(field);
-        if field_entry.field_type() != &FieldType::HierarchicalFacet {
-            return Err(crate::TantivyError::InvalidArgument(format!(
+
```
Comment (fmassot) [inline src/core/segment_reader.rs:111]: +1

**Review by fmassot — COMMENTED** [review]
@fulmicoton I'm ok with the PR. I'm just a little bit annoyed by the verbosity of tests but it's probably better to handle that in another issue.

**Pull request merged**